modules/directives/src/main/scala/scala/build/preprocessing/directives/DirectiveHandler.scala

```diff
@@ -15,10 +27,136 @@ trait DirectiveHandler[T] {
   def keys: Seq[String]
 
   def handleValues(
-    directive: StrictDirective,
-    path: Either[String, os.Path],
-    cwd: ScopePath,
+    scopedDirective: ScopedDirective,
     logger: Logger
   ): Either[BuildException, ProcessedDirective[T]]
 
+  /** It checks if the values assigned to the key of the scopedDirective have the expected type or
+    * number.
+    *
+    * Note: the correct functioning of this method for some of the handlers might require overriding
+    * the implementations for [[getSupportedTypes]], [[getValueNumberBounds]], and
```

Comment:
This is really superb pice of scaladoc - provides just the information needed - I really like it.